media/js/mozorg/home/home-b.js

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function($, Mozilla, dataLayer) {
+    'use strict';
+
+    var HomePageB = {
+        body: document.getElementById('intro-promo-content-wrapper'),
+        dataLayer: dataLayer,
+        state: 'Unknown',
+        syncCapable: false
+    };
+
+    HomePageB.setBodyClass = function(stateClass) {
+        var classes = HomePageB.body.className;
+        classes = classes.replace('state-default', stateClass);
+        HomePageB.body.className = classes;
+    };
+
+    HomePageB.trackPageState = function(state) {
+        HomePageB.dataLayer.push({
+            'event': 'page-load',
+            'browser': state
+        });
+    };
+
+    HomePageB.ctaSyncClick = function(e) {
+        e.preventDefault();
+
+        HomePageB.dataLayer.push({
+            'event': 'sync-click',
+            'browser': HomePageB.state
+        });
+
+        // TODO: do we need to set custom utm params here?
+        Mozilla.UITour.showFirefoxAccounts(HomePageB.params.utmParamsFxA());
+    };
+
+    HomePageB.init = function(config) {
+        HomePageB.client = config.client;
+        HomePageB.fxVersion = HomePageB.client.FirefoxMajorVersion;
+        HomePageB.params = new window._SearchParams();
```

Comment:
Hmm so, not necessarily introduced by this PR. But I noticed `'js/base/search-params.js'` is only available on all pages becuase it's part of the stub-attribution bundle. Given that can be toggled on/off independently, would it be better to move `'js/base/search-params.js'` to the common bundle?

Reply:
Good catch. I'll move it from the `stub-attribution` bundle to the `common` bundle.